### PR TITLE
feat(@desktop/wallet): Glue for sending nft

### LINKED ISSUE #12128: Send the transaction

### PR DESCRIPTION
**Pull request by Khushboo-dev-cpp**
fixed #12128 

go PR : https://github.com/status-im/status-go/pull/3945

### What does the PR do

This PR is the glue needed in order to send NFT Erc721 from the wallet

### Affected areas

SendModal

### StatusQ checklist

- [ ] add documentation if necessary (new component, new feature)
- [ ] update sandbox app
  - in case of new component, add new component page
  - in case of new features, add variation to existing component page
  - nice to have: add it to the demo application as well
- [ ] test changes in both light and dark theme?

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

https://github.com/status-im/status-desktop/assets/60327365/3db492c1-5c90-4ef9-ac7b-c12c9d37d4f6

https://github.com/status-im/status-desktop/assets/60327365/f30e0a3a-f209-4c01-a77b-fdc7be1fee33

### CODE REVIEW

File: ui/imports/shared/views/NetworkSelector.qml [hunk 1]
```diff
@@ -119,7 +122,7 @@ Item {
                 isBridgeTx: root.isBridgeTx
                 errorType: root.errorType
                 weiToEth: function(wei) {
-                    if(selectedAsset !== undefined)
+                    if(!!selectedAsset && selectedAsset !== undefined)
```
Comment (dlipicar) [inline ui/imports/shared/views/NetworkSelector.qml:125]: shouldn't the first condition be enough?

File: src/app/modules/main/wallet_section/send/network_model.nim [hunk 2]
```diff
@@ -275,6 +275,14 @@ QtObject:
         self.items[i].isEnabled = not disabled
         self.dataChanged(index, index, @[ModelRole.IsEnabled.int])
 
+  proc setEnabledFromChains*(self: NetworkModel, chainId: int) {.slot.} =
```
Comment (dlipicar) [inline src/app/modules/main/wallet_section/send/network_model.nim:278]: if the chains we can use are restricted, we should only display collectibles for those chains in the list. Could you update https://github.com/status-im/status-desktop/pull/12051/files#diff-1702286f2fd9311b35a8dcbc966e5d243ccd2d1dd351959048736a63c6710312R317 ?

File: ui/imports/shared/views/NetworkSelector.qml (hunk 1 above)
Comment (Khushboo-dev-cpp) [reply ui/imports/shared/views/NetworkSelector.qml:125]: both are needed else I get an undefined error

File: src/app/modules/main/wallet_section/send/network_model.nim (hunk 2 above)
Comment (Khushboo-dev-cpp) [reply src/app/modules/main/wallet_section/send/network_model.nim:278]: Hey I am only trying to set the chain the collectible is on to enabled=true
So this selection is based on what the collectible properties are and not vice versa, makes sense?

Comment (dlipicar) [reply src/app/modules/main/wallet_section/send/network_model.nim:278]: ohhh yeah that makes sense. Perhaps the method name is not specific enough in that case, maybe something like `setRouteEnabledFromChains`?

**Review by alaibe — APPROVED** [review]

File: src/app_service/service/transaction/service.nim [hunk 3]
```diff
@@ -232,14 +234,19 @@ QtObject:
       hopTx.amount = route.amountIn.some
       hopTx.bonderFee = route.bonderFees.some
       path.hopTx = hopTx
+    elif(route.bridgeName == ERC721_TRANSFER_NAME):
+      eRC721TransferTx = txData
+      eRC721TransferTx.chainID =  route.toNetwork.chainId.some
+      eRC721TransferTx.recipient = parseAddress(to_addr).some
+      eRC721TransferTx.tokenID = parseInt(tokenSymbol).some
```
Comment (dlipicar) [inline src/app_service/service/transaction/service.nim:241]: we shouldn't use `parseInt` and type `int` for this. TokenID is a `UInt256` variable, we should treat it like we treat the `value` field.

Comment (Khushboo-dev-cpp) [reply src/app_service/service/transaction/service.nim]: done @dlipicar !

**Pull request merged**